### DISCUSSION TITLE
schema is not validating properrly

### DISCUSSION THREAD

**Discussion by raghuk-tf** [fastapi/fastapi#5359] [Questions] [answered]
I have schema which is defined with 2 columns. How can i restrict request body .so that it will take only those 2 column. whenever  in the request if i pass a column twice its accepting.

**Comment by nikstuckenbrock** [comment]
Hey @raghuk-tf,

could you provide your schema? At the moment I'm not sure what you mean with ``columns``? :)

**Reply by raghuk-tf** [reply]
class student_schema(BaseModel):
    first_name: str
    last_name: str

    class Config:
        orm_mode = True

when i pass below json for post method :
{
"first_name": rag",
"first_name": "raghu",
"last_name": "kammala"
}

it is still accepting the record as below.
{
"first_name": "raghu",
"last_name": "kammala"
}
is there any way i can restrict that.
Thanks in Advance.

**Reply by nikstuckenbrock** [reply, marked as answer]
Hey @raghuk-tf,

I think this is not possible at the moment without customization. FastAPI uses [pydantic](https://pydantic-docs.helpmanual.io/) under the hood which then uses the default ``json`` [library](https://docs.python.org/3/library/json.html). The default ``json`` library produces the behavior you've described.

```python
import json
data = json.loads('{"first_name": "rag", "first_name": "raghu", "last_name": "kammala"}')
print(data)
```

This will result in:

```json
{"first_name": "raghu", "last_name": "kammala"}
```

This [thread](https://stackoverflow.com/questions/14902299/json-loads-allows-duplicate-keys-in-a-dictionary-overwriting-the-first-value#answer-14902564) describes the why and how to possibly fix it.
Let me know if this helps. :)
